.github/workflows/release.yaml

```diff
@@ -0,0 +1,127 @@
+name: Release pizza image
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+permissions:
+  actions: read
+  packages: write # for publish to ghcr.io
+  id-token: write # for signing image
+
+jobs:
+  build:
+    name: Build and publish image
+    runs-on: ubuntu-latest
+    env:
+      IMAGE_URI: ghcr.io/${{ github.repository }}
+      IMAGE_URI_TAG: ghcr.io/${{ github.repository }}:${{ github.ref_name }}
+    outputs:
+      image: ${{ env.IMAGE_URI }}
+      digest: ${{ steps.image_digest.outputs.IMAGE_DIGEST }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # tag=v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a # v2.1.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      
+      - name: Cross build
+        run: |
+          #!/usr/bin/env bash
+
+          docker buildx bake \
+            -f ./docker-bake.hcl \
+            --set build.args.GO_LDFLAGS="$GO_LDFLAGS" \
+            --set cross.tags=ghcr.io/${{ github.repository }}:${{ github.ref_name }} \
+            --push \
+            cross
+
+      - name: Install crane
+        if: startsWith(github.ref, 'refs/tags/')
```

Comment:
Are these `if` checks redundant since we are only running this action on tag pushes anyways?

```yaml
on:
  push:
    tags:
      - 'v*'
```

Reply:
I think it's ok to remove this

Reply:
On second thought, i think we may want to keep it, in case we want to run this pipeline even on main or manual trigger? what do u think?

Reply:
I think that sounds good!